File: src/net/minecraft/src/GuiAnimationSettingsOF.cpp
#include "GuiAnimationSettingsOF.h"
#include "GuiSmallButton.h"
#include "GuiSlider.h"
#include "GuiButton.h"
#include "GameSettings.h"
#include "EnumOptions.h"
#include "StringTranslate.h"
#include "ScaledResolution.h"
#include "Minecraft.h"

GuiAnimationSettingsOF::GuiAnimationSettingsOF(GuiScreen *parent, GameSettings *gamesettings)
	: title("Animation Settings")
	, prevScreen(parent)
	, settings(gamesettings)
{
}

void GuiAnimationSettingsOF::initGui()
{
	EnumOptions *enumOptions[] = {
		EnumOptions::ANIMATED_WATER, EnumOptions::ANIMATED_LAVA, EnumOptions::ANIMATED_FIRE,
		EnumOptions::ANIMATED_PORTAL, EnumOptions::ANIMATED_REDSTONE, EnumOptions::ANIMATED_EXPLOSION,
		EnumOptions::ANIMATED_FLAME, EnumOptions::ANIMATED_SMOKE, EnumOptions::VOID_PARTICLES,
		EnumOptions::WATER_PARTICLES, EnumOptions::RAIN_SPLASH, EnumOptions::PORTAL_PARTICLES,
		EnumOptions::PARTICLES, EnumOptions::DRIPPING_WATER_LAVA, EnumOptions::ANIMATED_TERRAIN,
		EnumOptions::ANIMATED_ITEMS, EnumOptions::ANIMATED_TEXTURES
	};

	StringTranslate *tr = StringTranslate::getInstance();
	int_t i = 0;
	for (EnumOptions *opt : enumOptions)
	{
		int_t x = width / 2 - 155 + i % 2 * 160;
		int_t y = height / 6 + 21 * (i / 2) - 10;
		if (!opt->getEnumFloat())
			controlList.push_back(new GuiSmallButton(opt->returnEnumOrdinal(), x, y, opt, settings->getKeyBinding(opt)));
		else
			controlList.push_back(new GuiSlider(opt->returnEnumOrdinal(), x, y, opt, settings->getKeyBinding(opt), settings->getOptionFloatValue(opt)));
		i++;
	}
	const int_t footerY = height / 6 + 168 + 11;
	controlList.push_back(new GuiButton(210, width / 2 - 155, footerY, 70, 20, "All ON"));
	controlList.push_back(new GuiButton(211, width / 2 - 75, footerY, 70, 20, "All OFF"));
	controlList.push_back(new GuiButton(200, width / 2 + 5, footerY, 150, 20, tr->translateKey("gui.done")));
}

void GuiAnimationSettingsOF::actionPerformed(GuiButton *button)
{
	if (!button->enabled) return;

	if (button->id < 100 && dynamic_cast<GuiSmallButton *>(button))
	{
		settings->setOptionValue(static_cast<GuiSmallButton *>(button)->returnEnumOptions(), 1);
		button->displayString = settings->getKeyBinding(EnumOptions::getEnumOptions(button->id));
	}
	if (button->id == 200)
	{
		mc->gameSettings->saveOptions();
		mc->displayGuiScreen(prevScreen);
		return;
	}
	if (button->id == 210)
		settings->setAllAnimations(true);
	if (button->id == 211)
		settings->setAllAnimations(false);
	if (button->id != EnumOptions::CLOUD_HEIGHT->returnEnumOrdinal())
	{
		ScaledResolution sr(mc->gameSettings, mc->displayWidth, mc->displayHeight);
		setWorldAndResolution(mc, sr.getScaledWidth(), sr.getScaledHeight());
	}
}

void GuiAnimationSettingsOF::drawScreen(int_t mouseX, int_t mouseY, float_t partialTick)
{
	drawDefaultBackground();
	drawCenteredString(fontRenderer, title, width / 2, 20, 0xffffff);
	GuiScreen::drawScreen(mouseX, mouseY, partialTick);
}

File: src/OptiFine_1.2.5_HD_C6.zip.src/GuiAnimationSettingsOF.java
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ public class GuiAnimationSettingsOF
/*     */   extends vp
/*     */ {
/*     */   private vp prevScreen;
/*     */   protected String title;
/*     */   private hu settings;
/*     */   
/*     */   public GuiAnimationSettingsOF(vp guiscreen, hu gamesettings) {
/*  19 */     this.title = "Animation Settings";
/*  20 */     this.prevScreen = guiscreen;
/*  21 */     this.settings = gamesettings;
/*     */   }
/*     */ 
/*     */   
/*     */   public void c() {
/*  26 */     adn stringtranslate = adn.a();
/*  27 */     int i = 0;
/*  28 */     zq[] aenumoptions = enumOptions;
/*  29 */     int j = aenumoptions.length;
/*  30 */     for (int k = 0; k < j; k++) {
/*     */       
/*  32 */       zq enumoptions = aenumoptions[k];
/*     */       
/*  34 */       int x = this.q / 2 - 155 + i % 2 * 160;
/*  35 */       int y = this.r / 6 + 21 * i / 2 - 10;
/*     */ 
/*     */       
/*  38 */       if (!enumoptions.a()) {
/*     */         
/*  40 */         this.s.add(new z(enumoptions.c(), x, y, enumoptions, this.settings.c(enumoptions)));
/*     */       } else {
/*     */         
/*  43 */         this.s.add(new ajz(enumoptions.c(), x, y, enumoptions, this.settings.c(enumoptions), this.settings.a(enumoptions)));
/*     */       } 
/*  45 */       i++;
/*     */     } 
/*     */ 
/*     */ 
/*     */     
/*  50 */     this.s.add(new abp(210, this.q / 2 - 155, this.r / 6 + 168 + 11, 70, 20, "All ON"));
/*  51 */     this.s.add(new abp(211, this.q / 2 - 155 + 80, this.r / 6 + 168 + 11, 70, 20, "All OFF"));
/*     */     
/*  53 */     this.s.add(new z(200, this.q / 2 + 5, this.r / 6 + 168 + 11, stringtranslate.b("gui.done")));
/*     */   }
/*     */ 
/*     */ 
/*     */   
/*     */   protected void a(abp guibutton) {
/*  59 */     if (!guibutton.h) {
/*     */       return;
/*     */     }
/*     */     
/*  63 */     if (guibutton.f < 100 && guibutton instanceof z) {
/*     */       
/*  65 */       this.settings.a(((z)guibutton).a(), 1);
/*  66 */       guibutton.e = this.settings.c(zq.a(guibutton.f));
/*     */     } 
/*  68 */     if (guibutton.f == 200) {
/*     */       
/*  70 */       this.p.A.b();
/*  71 */       this.p.a(this.prevScreen);
/*     */     } 
/*     */     
/*  74 */     if (guibutton.f == 210)
/*     */     {
/*  76 */       this.p.A.setAllAnimations(true);
/*     */     }
/*     */     
/*  79 */     if (guibutton.f == 211)
/*     */     {
/*  81 */       this.p.A.setAllAnimations(false);
/*     */     }
/*  83 */     if (guibutton.f != zq.CLOUD_HEIGHT.ordinal()) {
/*     */       
/*  85 */       agd scaledresolution = new agd(this.p.A, this.p.d, this.p.e);
/*  86 */       int i = scaledresolution.a();
/*  87 */       int j = scaledresolution.b();
/*  88 */       a(this.p, i, j);
/*     */     } 
/*     */   }
/*     */ 
/*     */   
/*     */   public void a(int i, int j, float f) {
/*  94 */     k();
/*  95 */     a(this.u, this.title, this.q / 2, 20, 16777215);
/*  96 */     super.a(i, j, f);
/*     */   }
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */ 
/*     */   
/* 106 */   private static zq[] enumOptions = new zq[] { zq.ANIMATED_WATER, zq.ANIMATED_LAVA, zq.ANIMATED_FIRE, zq.ANIMATED_PORTAL, zq.ANIMATED_REDSTONE, zq.ANIMATED_EXPLOSION, zq.ANIMATED_FLAME, zq.ANIMATED_SMOKE, zq.VOID_PARTICLES, zq.WATER_PARTICLES, zq.RAIN_SPLASH, zq.PORTAL_PARTICLES, zq.q, zq.DRIPPING_WATER_LAVA, zq.ANIMATED_TERRAIN, zq.ANIMATED_ITEMS, zq.ANIMATED_TEXTURES };
/*     */ }


/* Location:              D:\MC\OptiFine_1.2.5_HD_C6.zip!\GuiAnimationSettingsOF.class
 * Java compiler version: 5 (49.0)
 * JD-Core Version:       1.1.3
 */